Updating docs to explain how composition merges types under the hood (union and intersection) (#1839)
apollographql/federation: docs/source/federated-types/sharing-types.mdx

</CodeColumns>

> See [Input types and field arguments](./composition#input-types-and-field-arguments) to learn more.

Reviewer: ```suggestion
For more information, see [Input types and field arguments](./composition#input-types-and-field-arguments).
```

```

This can be useful when different subgraphs are responsible for different subsets of a particular set of related types or values.

> >  You can also use the `enum` type across multiple subgraphs, see [Merging types from multiple subgraphs](composition#enums) to learn more.

Reviewer: ```suggestion
>  You can also use the `enum` type across multiple subgraphs. For details, see [Merging types from multiple subgraphs](composition#enums).
```
